app/src/main/java/org/schabi/newpipe/settings/tabs/Tab.java

```diff
@@ -576,15 +600,34 @@ protected void readDataFromJson(final JsonObject jsonObject) {
 
         @Override
         public boolean equals(final Object obj) {
-            if (!(super.equals(obj)
-                    && Objects.equals(playlistType, ((PlaylistTab) obj).playlistType)
-                    && Objects.equals(playlistName, ((PlaylistTab) obj).playlistName))) {
-                return false; // base objects are different
+            if (!(obj instanceof PlaylistTab)) {
+                return false;
+            }
+
+            final PlaylistTab other = (PlaylistTab) obj;
+
+            if (!(super.equals(obj))) {
+                // Base objects are different
+                return false;
             }
 
-            return (playlistId == ((PlaylistTab) obj).playlistId)                     // local
-                    || (playlistServiceId == ((PlaylistTab) obj).playlistServiceId    // remote
-                    && Objects.equals(playlistUrl, ((PlaylistTab) obj).playlistUrl));
+            return playlistServiceId == other.playlistServiceId // Remote
```

Comment:
Let's keep it consistent
```suggestion
            return super.equals(obj)
                    && playlistServiceId == other.playlistServiceId // Remote
```